spec/Section 6 -- Execution.md

```diff
@@ -564,39 +564,26 @@ Each field requested in the grouped field set that is defined on the selected
 objectType will result in an entry in the response map. Field execution first
 coerces any provided argument values, then resolves a value for the field, and
 finally completes that value either by recursively executing another selection
-set or coercing a scalar value. `ccnPropagationPairs` is an unordered map where
-the keys are paths of required fields, and values are paths of the nearest
-optional parent to those required fields. `currentPropagationPath` starts as an
-empty path to indicate that `null` propagation should continue until it hits
-`data` if there is no optional field.
+set or coercing a scalar value.
 
-ExecuteField(objectType, objectValue, fieldType, fields, variableValues,
-currentPropagationPath, ccnPropagationPairs):
+ExecuteField(objectType, objectValue, fieldType, fields, variableValues):
 
 - Let {field} be the first entry in {fields}.
 - Let {fieldName} be the field name of {field}.
-- Let {requiredStatus} be the required status of {field}.
-- Let {newPropagationPath} be {path} if {requiredStatus} is optional, otherwise
-  let {newPropagationPath} be {currentPropagationPath}
-- If {requiredStatus} is optional:
-  - Let {newPropagationPath} be {path}
-- If {requiredStatus} is required:
-  - Set {path} to {newPropagationPath} in {ccnPropagationPairs}
 - Let {argumentValues} be the result of {CoerceArgumentValues(objectType, field,
   variableValues)}
 - Let {resolvedValue} be {ResolveFieldValue(objectType, objectValue, fieldName,
   argumentValues)}.
-- Let {modifiedFieldType} be {ApplyRequiredStatus(fieldType, requiredStatus)}.
-- Return the result of {CompleteValue(modifiedFieldType, fields, resolvedValue,
-  variableValues, newPropagationPath, ccnPropagationPairs)}.
+- Return the result of {CompleteValue(fieldType, fields, resolvedValue,
+  variableValues)}.
 
 ## Accounting For Client Controlled Nullability Designators
 
 A field can have its nullability status set either in its service's schema, or a
-nullability designator (`!` or `?`) can override it for the duration of an
-execution. In order to determine a field's true nullability, both are taken into
-account and a final type is produced. A field marked with a `!` is called a
-"required field" and a field marked with a `?` is called an optional field.
+nullability designator (`!`) can override it for the duration of an execution.
+In order to determine a field's true nullability, both are taken into account
+and a final type is produced. A field marked with a `!` is called a "required
+field".
 
 ApplyRequiredStatus(type, requiredStatus):
```

Comment:
I think I got rid of `ApplyRequiredStatus` from the execution PR. I think the new function that matters here is called `simpleTypeTransfer` or something like that.

Reply:
I removed `ApplyRequiredStatus`, defined `simpleTypeTransfer`, and added references to `simpleTypeTransfer`. Can you review the `list` case of `ComputeValue`?